static/app/components/performance/spanSearchQueryBuilder.tsx

```diff
@@ -8,39 +8,88 @@ import {t} from 'sentry/locale';
 import type {PageFilters} from 'sentry/types/core';
 import {SavedSearchType, type Tag, type TagCollection} from 'sentry/types/group';
 import {defined} from 'sentry/utils';
-import {isAggregateField, isMeasurement} from 'sentry/utils/discover/fields';
-import {DEVICE_CLASS_TAG_VALUES, isDeviceClass} from 'sentry/utils/fields';
+import {
+  isAggregateField,
+  isEquation,
+  isMeasurement,
+  parseFunction,
+} from 'sentry/utils/discover/fields';
+import {
+  DEVICE_CLASS_TAG_VALUES,
+  FieldKind,
+  getFieldDefinition,
+  isDeviceClass,
+} from 'sentry/utils/fields';
 import useApi from 'sentry/utils/useApi';
 import useOrganization from 'sentry/utils/useOrganization';
 import usePageFilters from 'sentry/utils/usePageFilters';
+import {isCustomMeasurement} from 'sentry/views/dashboards/utils';
 import {SPANS_FILTER_KEY_SECTIONS} from 'sentry/views/insights/constants';
 import {
   useSpanFieldCustomTags,
   useSpanFieldSupportedTags,
 } from 'sentry/views/performance/utils/useSpanFieldSupportedTags';
 
+import {STATIC_FIELD_TAGS_SET} from '../events/searchBarFieldConstants';
+
 interface SpanSearchQueryBuilderProps {
   initialQuery: string;
   searchSource: string;
+  allowAggregateFunctions?: boolean;
   datetime?: PageFilters['datetime'];
   disableLoadingTags?: boolean;
+  fields?: string[];
   onSearch?: (query: string, state: CallbackSearchState) => void;
   placeholder?: string;
   projects?: PageFilters['projects'];
 }
 
+const getFunctionTags = (fields: string[]) => {
+  if (!fields?.length) {
+    return [];
+  }
+  return fields.reduce((acc, item) => {
+    if (
+      !STATIC_FIELD_TAGS_SET.has(item) &&
+      !isEquation(item) &&
+      !isCustomMeasurement(item)
+    ) {
+      const parsedFunction = parseFunction(item);
+      if (parsedFunction) {
+        acc[parsedFunction.name] = {
+          key: parsedFunction.name,
+          name: parsedFunction.name,
+          kind: FieldKind.FUNCTION,
+        };
+      }
+    }
+
+    return acc;
+  }, {});
+};
+
+const getSpanFieldDefinition = (key: string) => {
+  return getFieldDefinition(key, 'span');
+};
+
 export function SpanSearchQueryBuilder({
   initialQuery,
   searchSource,
   datetime,
   onSearch,
   placeholder,
   projects,
+  fields = [],
+  allowAggregateFunctions,
```

Comment:
I think you can simplify this further and just pass in a list of supported aggregation functions.

So the parent component can simply choose to pass in the list instead of having it here